src/Libraries/DSCPython/Encoders/ListEncoder.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Dynamo.Utilities;
+using Python.Runtime;
+
+namespace DSCPython.Encoders
+{
+    internal class ListEncoder : IPyObjectEncoder, IPyObjectDecoder
+    {
+        private static readonly Type[] decodableTypes = new Type[]
+        {
+            typeof(IList), typeof(ArrayList),
+            typeof(IList<>), typeof(List<>)
+        };
+
+        public bool CanDecode(PyObject objectType, Type targetType)
+        {
+            if (targetType.IsGenericType)
+            {
+                targetType = targetType.GetGenericTypeDefinition();
+            }
+            return decodableTypes.IndexOf(targetType) >= 0;
+        }
+
+        public bool CanEncode(Type type)
+        {
+            return typeof(IList).IsAssignableFrom(type);
+        }
+
+        public bool TryDecode<T>(PyObject pyObj, out T value)
+        {
+            if (!PySequence.IsSequenceType(pyObj))
+            {
+                value = default;
+                return false;
+            }
+
+            using (var pyList = PyList.AsList(pyObj))
```

Comment:
I'm confused by this using statement, does `AsList` return a .net list or a Python List?

Reply:
It's a Python list